src/figuresClasses.ts

```diff
@@ -1,19 +1,99 @@
-export interface Figure {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number,
+}
 
+function checkSides(...args: number[]): void {
+  if (Math.min(...args) <= 0) {
+    throw new Error('The length cannot to be negative number');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  readonly shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkSides(a, b, c);
+
+    const [
+      sideA,
+      sideB,
+      sideC,
+    ]: number[] = [a, b, c].sort((prev, next) => prev - next);
+
+    if (sideC >= sideB + sideA) {
+      throw new Error('Impossible to build triangle with these sides!');
+    }
+  }
 
+  getArea(): number {
+    const semiperimeter: number = (this.a + this.b + this.c) / 2;
+    const area: number = Math.sqrt(
+      semiperimeter
+      * (semiperimeter - this.a)
+      * (semiperimeter - this.b)
+      * (semiperimeter - this.c),
+    );
+
+    return +(area.toFixed(2));
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  readonly shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Wrong length of radius');
+    }
```

Comment:
here you can check the already created function checkSides.

create an abstract class that will implement Figure and add a function to it.

```

abstract class BasedForm implements Figure {
  abstract shape: Shape;

  abstract color: Color;

  abstract getArea(): number;

  protected checkAllSides(sides: number[]): void {
    if (sides.some((value: number) => value <= 0)) {
      throw new Error(
        `figure of shape ${this.shape} should have only positive params`,
      );
    }
  }
}
```